refactor: split some contents from quick-start (#381)
nervosnetwork/docs.nervos.org: website/docs/sdk-and-devtool/cli-tool.mdx

---
id: cli-tools
title: CLI Tools
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";
import Tooltip from "@components/Tooltip";

> # CLI Tools

Reviewer: Should we separate CKB-CLI and OffCKB to different pages (Just like what we do for the SDKs)?

Author: I'll give it a thought. let's leave them under cli for now.

> Commandline tools are often used by developers to perform specific blockchain tasks in a faster and more efficient way.
> 
> ## CKB-CLI
> 
> [ckb-cli](https://github.com/nervosnetwork/ckb-cli) is a CKB command line tool developed in Rust Lang.
> 
> If you need to interact with Mainnet blockchain from commandline, `ckb-cli` is a ideal choice to manager accounts, deploy Scripts and send Transactions.

Reviewer: ```suggestion
Command-line tools are often used by developers to perform specific blockchain tasks in a faster and more efficient way.

## CKB-CLI

[ckb-cli](https://github.com/nervosnetwork/ckb-cli) is a CKB command-line tool developed in Rust.

If you need to interact with Mainnet blockchain from the command line, `ckb-cli` is an ideal choice to manager accounts, deploy Scripts and send Transactions.
```

### Features

```
    rpc         Invoke RPC call to node
    account     Manage accounts
    mock-tx     Handle mock transactions (verify/send)
    tx          Handle common sighash/multisig transaction
    util        Utilities
    molecule    Molecule encode/decode utilities
    wallet      Transfer / query balance (with local index) / key utils
    dao         Deposit / prepare / withdraw / query NervosDAO balance (with local index) / key utils
```

> All second level sub-commands are listed in [wiki page](https://github.com/nervosnetwork/ckb-cli/wiki/Sub-Commands).

Reviewer: ```suggestion
All second-level sub-commands are listed on the [wiki page](https://github.com/nervosnetwork/ckb-cli/wiki/Sub-Commands).
```

### Install

```
git clone https://github.com/nervosnetwork/ckb-cli.git
cd ckb-cli
cargo install --path . -f --locked
```

### Usage

Better export an env first (or give in argument)

```
export API_URL=http://127.0.0.1:8114
```

Directly go to **gorgeous** interactive mode:

```
ckb-cli
```

Show available commands

```shell
# Top level help doc
ckb-cli --help
# RPC help doc
ckb-cli rpc --help
```

> #### Example: Get live cell (json output format)

Reviewer: ```suggestion
#### Example: Get Live Cell (JSON Output Format)
```

```
ckb-cli rpc get_live_cell --tx-hash 0x4ec75b5a8de8d180853d5046760a99285c73283a5dc528f81d6ee056f5335172 --index 0
```

**Response:**

```json
{
  "cell": {
    "capacity": "125000000000",
    "lock": {
      "args": ["0x64257f00b6b63e987609fa9be2d0c86d351020fb"],
      "code_hash": "0x1892ea40d82b53c678ff88312450bbb17e164d7a3e0a90941aa58839f56f8df2",
      "hash_type": "type"
    },
    "type": null
  },
  "status": "live"
}
```

## OffCKB

> [offckb](https://github.com/RetricSu/offckb) is a CLI tool developed in Nodejs to help you quickly set up a predefined CKB Devnet and create dApp boilerplate.

Reviewer: ```suggestion
[offckb](https://github.com/RetricSu/offckb) is a CLI tool developed in Node.js to help you quickly set up a predefined CKB Devnet and create dApp boilerplates.
```

  node                                              Use the CKB to start devnet
  clean                                             Clean the devnet data, need to stop running the chain first
  accounts                                          Print account list info
  list-hashes                                       Use the CKB to list blockchain scripts hashes
  inject-config                                     Add offckb.config.ts to your workspace
  sync-config                                       sync offckb.config.ts in your workspace
  deposit [options] [toAddress] [amountInShannon]   Deposit CKB tokens to address, only devnet and testnet
  transfer [options] [toAddress] [amountInShannon]  Transfer CKB tokens to address, only devnet and testnet
  balance [options] [toAddress]                     Check account balance, only devnet and testnet
  deploy [options]                                  Deploy contracts to different networks, only supports devnet and testnet
  deployed-scripts [options]                        Show deployed contracts info on networks, only supports devnet and testnet
  help [command]                                    display help for command
```

> #### Create a full-stack Project

Reviewer: ```suggestion
#### Create a Full-Stack Project
```

Create a new project from predefined boilerplates.

```sh
offckb create <your-project-name, eg:my-first-ckb-project>
```

> The boilerplate can be targeting on different CKB networks. Check [README.md](https://github.com/nervosnetwork/docs.nervos.org/blob/develop/examples/remix-vite-template/readme.md) in the project to get started.

Reviewer: ```suggestion
The boilerplate can target on different CKB networks. Check the [README.md](https://github.com/nervosnetwork/docs.nervos.org/blob/develop/examples/remix-vite-template/readme.md) in the project to get started.
```
